etc/ncp-templates/nextcloud.conf.sh

```diff
@@ -0,0 +1,85 @@
+#! /bin/bash
+
+set -e
+source /usr/local/etc/library.sh
+source "${BINDIR}/NETWORKING/letsencrypt.sh"
+
+if [[ "$DOCKERBUILD" == 1 ]]
+then
+  source "${BINDIR}/SYSTEM/metrics.sh"
+else
+  tmpl_metrics_enabled(){ return 1; }
+fi
+
+echo "### DO NOT EDIT! THIS FILE HAS BEEN AUTOMATICALLY GENERATED. CHANGES WILL BE OVERWRITTEN ###"
+echo ""
+
+cat <<EOF
+<IfModule mod_ssl.c>
+  <VirtualHost _default_:443>
+    DocumentRoot /var/www/nextcloud
+EOF
+
+LETSENCRYPT_DOMAIN="$(tmpl_letsencrypt_domain || true)"
+if [[ "$1" != "--defaults" ]] && [[ -n "$LETSENCRYPT_DOMAIN" ]]
+then
+  echo "    ServerName ${LETSENCRYPT_DOMAIN}"
+  LETSENCRYPT_CERT_BASE_PATH="/etc/letsencrypt/live/${LETSENCRYPT_DOMAIN,,}"
+  LETSENCRYPT_CERT_PATH="${LETSENCRYPT_CERT_BASE_PATH}/fullchain.pem"
+  LETSENCRYPT_KEY_PATH="${LETSENCRYPT_CERT_BASE_PATH}/privkey.pem"
+else
+  unset LETSENCRYPT_DOMAIN
+fi
+
+cat <<EOF
+    CustomLog /var/log/apache2/nc-access.log combined
+    ErrorLog  /var/log/apache2/nc-error.log
+    SSLEngine on
+    SSLCertificateFile      ${LETSENCRYPT_CERT_PATH:-/etc/ssl/certs/ssl-cert-snakeoil.pem}
```

Comment:
hmmm this could automatically solve https://github.com/nextcloud/nextcloudpi/pull/1254

We should make sure to generate the certs if they don't exist though

Reply:
True :smiley:

Reply:
But shouldn't they exist anyways, since they are used during initialization?

Reply:
sure, but if there is a way to break it there will be a user that will. That's one lesson I learned doing this :)

Reply:
Okay, will do :D